src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -295,6 +295,10 @@ bool java_bytecode_languaget::generate_support_functions(
   if(!res.is_success())
     return res.is_error();
 
+  // Load the main function into the symbol table to get access to its
+  // parameter names
+  convert_lazy_method(res.main_function.name, symbol_table);
```

Comment:
I'm afraid I don't understand the commit subject. None of the changes here seem to be doing anything in `main`.

Reply:
The line you're commenting on says `main_function` in the middle. That's the one we're faulting in.

Reply:
As English isn't my first language: what does "faulting" mean here?

Reply:
Sorry for the confusion. Think of it like a page fault in a memory mapper. We're converting the main function and adding it to the cache of loaded functions.

Reply:
Ok, I was guessing along those lines, but if you can find a way of wording the commit message so that even simple-minded people like myself can grasp it this would be appreciated.